prime-dotnet-webapi/Models/Api/PaginatedList.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Prime.Models.Api
+{
+    public class PaginatedResponse<T>
+    {
+        public List<T> Results { get; set; }
+        public int Page { get; set; }
+        public int TotalPages { get; set; }
+        public int Total { get; set; }
+        public int PageSize { get; set; }
+
+        public PaginatedResponse(PaginatedList<T> list)
+        {
+            Results = list;
+            Page = list.Page;
+            TotalPages = list.TotalPages;
+            Total = list.Total;
+            PageSize = list.PageSize;
+        }
+    }
+    public class PaginatedList<T> : List<T>
+    {
+        private static int GetPageSize()
+        {
+            return 100;
+        }
+
+        public int Page { get; private set; }
+        public int TotalPages { get; private set; }
+        public int Total { get; private set; }
+        public int PageSize { get; private set; }
+
+        public PaginatedList(List<T> items, int count, int page, int pageSize)
+        {
+            Page = page;
```

Comment:
It works well and this is not a normal use case but I can put any number in the query param. For example I have 300 enrollees which equals 3 pages, but I can put `page=6` in the url then I am taken to an empty page and I can still go back and it decreases properly through pages 5 and 4 then 3 which has enrollees. We also get an error for a negative page number. We can throw the user back to the last page instead or first page if they enter a page number outside the range. Again not a normal scenario.

Reply:
So no we have the negative page number case solved that it sends me to the first page, but I still can enter a page number beyond the range and it would still take me to an empty page. that's my only feedback but I think everything works great!

Reply:
yeah indexing outside of the pages doesn't break anything and is obvious to the user what is going on. Looked at some other sites that use pagination and there doesn't seem like an accepted solution. Even the negative number some would return no results, send you to first page, or error out. I feel if a user edits the url and goes to a page with no items they can easily navigate to the results using the tools on the page.